lib/path/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![deny(warnings)]
```

Comment:
Are we working on moving this up to apply to everything in the project?

Reply:
I'm not really convinced either way yet on this. I think it makes sense on binaries where the dependencies are locked, but for a library, you can have dependency versions change at any time that could start to generate warnings and fail CI. I'm just keeping this as-is for now until it's more clear how to proceed here.

Reply:
Fair enough, I was definitely thinking of things from how we've been doing Vector